### PR TITLE
Bugfix/#4451/Fixed file uploading

### PR DESCRIPTION
**Pull request by zaitsev-oleksii**
Issue: https://github.com/ita-social-projects/GreenCity/issues/4451

### CODE REVIEW

File: src/app/ubs/ubs-admin/components/add-violations/add-violations.component.html [hunk 1]
```diff
@@ -49,41 +49,53 @@ <h6>{{ 'add-violation-modal.label-for-photo-container' | translate }}</h6>
...
         </div>
-        <p class="drag-inscription">{{ img.label }}</p>
+      </div>
+      <div
+        class="image-container"
+        appDrag
+        (files)="onFilesDropped($event)"
+        *ngFor="let placeholder of maxNumberOfImgs - images.length | repeat; index as i"
+      >
+        <p class="drag-inscription" *ngIf="i === 0">{{ 'add-violation-modal.drag-photo' | translate }}</p>
```
Comment (hnativlyubomyr) [inline src/app/ubs/ubs-admin/components/add-violations/add-violations.component.html]: *ngIf="!i"

File: src/app/ubs/ubs-admin/components/add-violations/add-violations.component.ts [hunk 2]
```diff
@@ -23,28 +22,28 @@ interface DataToSend {
   imagesToDelete?: string[] | null;
 }
 
+interface ViolationImage {
+  src: string;
+  name: string | null;
+  file: File;
+}
+
```
Comment (hnativlyubomyr) [inline src/app/ubs/ubs-admin/components/add-violations/add-violations.component.ts:30]: bad practice, interfaces should be move out of the component

File: src/app/ubs/ubs-admin/components/add-violations/add-violations.component.ts [hunk 3]
```diff
@@ -101,59 +95,53 @@ export class AddViolationsComponent implements OnInit, OnDestroy {
...
-          this.imagesFromDBLength = violation.images.length;
-          this.imagesFromDB = violation.images;
-          this.date = violation.violationDate;
-          this.isLoading = false;
-          this.initialData = {
-            violationLevel: violation.violationLevel,
-            violationDescription: violation.description,
-            initialImagesLength: violation.images.length
-          };
+  private loadInitialData() {
```
Comment (hnativlyubomyr) [inline src/app/ubs/ubs-admin/components/add-violations/add-violations.component.ts]: add return type - void

File: src/app/ubs/ubs-admin/components/add-violations/add-violations.component.ts [hunk 4]
```diff
@@ -101,59 +95,53 @@ export class AddViolationsComponent implements OnInit, OnDestroy {
...
-      data.imagesToDelete = this.deletedImages.length ? this.deletedImages : null;
+      data.imagesToDelete = this.imagesToDelete.length ? this.imagesToDelete : null;
     }
     const formData: FormData = new FormData();
     const stringifiedDataToSend = JSON.stringify(data);
-    formData.append(dto, stringifiedDataToSend);
-    for (const images of this.imgArray) {
-      formData.append(this.editMode ? 'multipartFiles' : 'files', images);
+    formData.append('add', stringifiedDataToSend);
+    for (const image of this.images) {
```
Comment (hnativlyubomyr) [inline src/app/ubs/ubs-admin/components/add-violations/add-violations.component.ts]: don't use for, this.images.forEach(...

File: src/app/ubs/ubs-admin/components/add-violations/add-violations.component.ts [hunk 5]
```diff
@@ -170,80 +158,71 @@ export class AddViolationsComponent implements OnInit, OnDestroy {
...
+      return;
+    }
+
     this.isImageSizeError = false;
     this.isImageTypeError = false;
-    this.checkFileExtension(files);
-    const imageFile = files[0].file;
-    this.transferFile(imageFile);
+
+    for (const file of files) {
```
Comment (hnativlyubomyr) [inline src/app/ubs/ubs-admin/components/add-violations/add-violations.component.ts]: the same

File: src/app/ubs/ubs-admin/components/add-violations/add-violations.component.ts [hunk 6]
```diff
@@ -170,80 +158,71 @@ export class AddViolationsComponent implements OnInit, OnDestroy {
...
+      }
+      if (!validSize) {
+        this.isImageSizeError = true;
+        continue;
+      }
+      this.transferFile(file);
+      if (this.editMode) {
+        this.isInitialImageDataChanged = true;
+      }
+    }
```
Comment (hnativlyubomyr) [inline src/app/ubs/ubs-admin/components/add-violations/add-violations.component.ts]: also remove continue:
this.isImageTypeError = !this.validateFileExtension(file);
this.isImageSizeError = !this.validateFileSize(file);

if (this.isImageTypeError || this.isImageSizeError) {
this.transferFile(file);
this.isInitialImageDataChanged = this.editMode;
}

**Review by hnativlyubomyr — APPROVED** [review]

File: src/app/ubs/ubs-admin/components/add-violations/add-violations.component.ts (hunk 2 above)
Comment (zaitsev-oleksii) [reply src/app/ubs/ubs-admin/components/add-violations/add-violations.component.ts:30]: It won't be needed anywhere else, should I remove it completely then?

**Review by vplvua — APPROVED** [review]

**Review by atsa21 — APPROVED** [review]

**Review by AngelinaRudnitska — APPROVED** [review]

**Review by marianaharnyk — APPROVED** [review]

**Pull request merged**